### PR TITLE
[firebase_admob] Change ad events to trigger similar to their platforms

### PR DESCRIPTION
**Pull request by bparrishMines**
## Description

**Breaking Change** Changes the name of `onAdClicked` and `onAdImpression` to `onNativeAdClicked` and `onNativeAdImpression` to keep clear that they only occur on `NativeAds`. This also changes `onApplicationLeft` to `onApplicationExit` since this can be called 
before the application leaves on iOS and after on Android. 

## Related Issues

see b/154007890

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_admob/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.9.3+4
+
+* TODO: Temp fix for test
```
Comment (blasten) [inline packages/firebase_admob/CHANGELOG.md:3]: q: is this TODO real?

File: packages/firebase_admob/pubspec.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: firebase_admob
 description: Flutter plugin for Firebase AdMob, supporting
   banner, interstitial (full-screen), and rewarded video ads
 homepage: https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_admob
-version: 0.9.3+3
+version: 0.9.3+4
```
Comment (blasten) [inline packages/firebase_admob/pubspec.yaml:5]: is this a real major or a patch version bump?

**Comment by blasten** [comment]
do we test this stuff somehow?

**Comment by bparrishMines** [comment]
Closing in favor of: https://github.com/FirebaseExtended/flutterfire/issues/2521

**Pull request closed without merging**